ygnmi/ygnmi_test.go

```diff
@@ -2226,6 +2226,138 @@ func TestDelete(t *testing.T) {
 	}
 }
 
+func TestBatchGet(t *testing.T) {
+	fakeGNMI, c := newClient(t)
+	aLeafStatePath := testutil.GNMIPath(t, "/remote-container/state/a-leaf")
+	twoPath := testutil.GNMIPath(t, "/parent/child/state/two")
+	aLeafSubath := testutil.GNMIPath(t, "/remote-container/*/a-leaf")
+
+	tests := []struct {
+		desc                 string
+		stub                 func(s *testutil.Stubber)
+		config               bool
+		paths                []ygnmi.PathStruct
+		wantSubscriptionPath []*gpb.Path
+		wantVal              *ygnmi.Value[*exampleoc.Root]
+		wantErr              string
+	}{{
+		desc: "state leaves",
+		stub: func(s *testutil.Stubber) {
+			s.Notification(&gpb.Notification{
+				Timestamp: 100,
+				Update: []*gpb.Update{{
+					Path: aLeafStatePath,
+					Val:  &gpb.TypedValue{Value: &gpb.TypedValue_StringVal{StringVal: "foo"}},
+				}, {
+					Path: twoPath,
+					Val:  &gpb.TypedValue{Value: &gpb.TypedValue_StringVal{StringVal: "bar"}},
+				}},
+			}).Sync()
+		},
+		paths: []ygnmi.PathStruct{
+			root.New().RemoteContainer().ALeaf(),
+			root.New().Parent().Child().Two(),
+		},
+		wantSubscriptionPath: []*gpb.Path{
+			aLeafSubath,
+			twoPath,
+		},
+		wantVal: (&ygnmi.Value[*exampleoc.Root]{
+			Timestamp: time.Unix(0, 100),
+			Path:      testutil.GNMIPath(t, "/"),
+		}).SetVal(&exampleoc.Root{
+			RemoteContainer: &exampleoc.RemoteContainer{ALeaf: ygot.String("foo")},
+			Parent:          &exampleoc.Parent{Child: &exampleoc.Parent_Child{Two: ygot.String("bar")}},
+		}),
+	}, {
+		desc:   "config ignore state leaves",
+		config: true,
+		stub: func(s *testutil.Stubber) {
+			s.Notification(&gpb.Notification{
+				Timestamp: 100,
+				Update: []*gpb.Update{{
+					Path: aLeafStatePath,
+					Val:  &gpb.TypedValue{Value: &gpb.TypedValue_StringVal{StringVal: "foo"}},
+				}, {
+					Path: twoPath,
+					Val:  &gpb.TypedValue{Value: &gpb.TypedValue_StringVal{StringVal: "bar"}},
+				}},
+			}).Sync()
+		},
+		paths: []ygnmi.PathStruct{
+			root.New().RemoteContainer().ALeaf(),
+			root.New().Parent().Child().Two(),
+		},
+		wantSubscriptionPath: []*gpb.Path{
+			aLeafSubath,
+			twoPath,
+		},
+		wantVal: (&ygnmi.Value[*exampleoc.Root]{
+			Timestamp: time.Unix(0, 100),
+			Path:      testutil.GNMIPath(t, "/"),
+		}).SetVal(&exampleoc.Root{
+			RemoteContainer: &exampleoc.RemoteContainer{},
+			Parent:          &exampleoc.Parent{Child: &exampleoc.Parent_Child{Two: ygot.String("bar")}},
```

Comment:
I wanted to call this out because its a little weird. If you add state-only path to a batch, and create a Config() query. We won't discard the state paths (because they have no shadow path). I changed to path struct generation to make that more obvious. Is behavior ok?

Reply:
Thanks for pointing this out. I'm thinking we should only send the paths we actually care about -- I think that's what the user is expecting. Could we resolve the Config/stateness of the paths when we're resolving the Batch? e.g. for a Config() batch query, filter out paths by whether they have a `Config()` method, and only if we do do we add to subpaths?

Reply:
I think we should figure out a way to fix this but I'm not sure of the easiest implementation.

Reply:
As discussed offline when we say "Batch.Config()" we're really saying "Batch.PreferConfig()", where we discard the state only if there is no space for it in the GoStruct. Could you try using `PruneConfigFalse()` to discard these state-only values?

Reply:
If it's gonna be some work I'm happy with pushing this to a later PR, I think these two things are separate concerns since it's an existing issue.

Reply:
it's pretty trivial, made the change. i'll leave whether we want to prune empty branches for another time though